scripts/ci-build.sh

```diff
@@ -0,0 +1,562 @@
+#!/bin/bash -e
+
+# (C) 2018 GENIVI Alliance
+# LICENSE: MPLv2
+
+# ---- General settings ----
+
+# Fixed cache location on all agents that support this.
+AGENT_STANDARD_DL_DIR="/var/cache/yocto/downloads"
+AGENT_STANDARD_SSTATE_DIR="/var/cache/yocto/sstate"
+AGENT_STANDARD_SGX_LOCATION="/var/go/sgx_bin"
+AGENT_STANDARD_SGX_GEN3_LOCATION="/var/go/sgx_bin_gen3"
+
+# ---- Helper functions ----
+
+fail=false
+
+# deref() is some trickery, but useful when a variable *name* itself is created
+# on-the-fly by the program, and then a value is looked for in that variable
+# name.
+
+# Return the value of a variable whose name is stored in another variable
+deref() { eval echo \$$1 ; }
+
+# For shell/environment variables, whether they are named explicitly in the
+# program, or the name is generated on the fly, this single function
+# encapsulate a check that there is a value defined for the variable.
+ensure_var_is_defined() {
+  var_name=$1
+  current_value=$(deref $var_name)
+  if [[ -z "$current_value" ]]; then
+    echo "Please define environment var_name \$$var_name (found empty value)"
+    fail=true  # We want all the errors reported, so don't halt yet.
+  fi
+}
+
+# This function takes a variable name and a "default" value If no value was
+# assigned to the variable already (e.g. from environment) the given "default"
+# value is assigned.
+define_with_default() {
+  var_name=$1
+  default="$2"
+  current_value=$(deref $var_name)
+  if [[ -z "$current_value" ]]; then
+    eval $var_name="$default"
+  fi
+}
+
+# Helper function to terminate the program on failure.
+# This allows us to delay program termination a bit by setting fail flag
+# precisely when the error is found, but a bit later on call the
+# stop_if_failure function
+stop_if_failure() {
+  if [[ "$fail" == "true" ]] ; then
+    exit 1
+  fi
+}
+
+# As the function says... (inform user and exit)
+stop_immediately() {
+  echo "Fatal error occurred - stopping ci-build script"
+  exit 2
+}
+
+# Append a given value to the project's bitbake local.conf file
+append_local_conf() {
+  LOCAL_CONF="$BASEDIR/build/conf/local.conf"
+  if [[ -f "$LOCAL_CONF" ]]; then
+    echo -n "Appending to local.conf: "
+    cat <<EOT | tee -a "$LOCAL_CONF"
+$1
+EOT
```

Comment:
The script will continue appending to an existing local.conf file for every time you try to run it. Ie, we wind up with the MIRRORS_append once every time we try to run this script as it is now.

Reply:
Agreed, this is because CI environments are thrown away frequently.  So this could be improved for end-user, yes.